commands/get/resources/getter.go

```diff
@@ -0,0 +1,145 @@
+package resources
+
+import (
+	"github.com/opencontrol/compliance-masonry/tools/constants"
+	"log"
+	"os"
+	"path/filepath"
+	"github.com/opencontrol/compliance-masonry/lib/common"
+	"github.com/opencontrol/compliance-masonry/lib/opencontrol"
+	"github.com/opencontrol/compliance-masonry/tools/mapset"
+	"github.com/opencontrol/compliance-masonry/tools/fs"
+)
+
+// GetResources will download all the resources that are specified by the schema first by copying the
+// local resources then downloading the remote ones and letting their respective schema version handle
+// how to get their resources.
+func GetResources(source string, destination string, opencontrol common.OpenControl, getter Getter) error {
+	// Local
+	// Get Certifications
+	log.Println("Retrieving certifications")
+	err := getter.GetLocalResources(source, opencontrol.GetCertifications(), destination,
+		constants.DefaultCertificationsFolder, false, constants.Certifications)
+	if err != nil {
+		return err
+	}
+	// Get Standards
+	log.Println("Retrieving standards")
+	err = getter.GetLocalResources(source, opencontrol.GetStandards(), destination,
+		constants.DefaultStandardsFolder, false, constants.Standards)
+	if err != nil {
+		return err
+	}
+	// Get Components
+	log.Println("Retrieving components")
+	err = getter.GetLocalResources(source, opencontrol.GetComponents(), destination,
+		constants.DefaultComponentsFolder, true, constants.Components)
+	if err != nil {
+		return err
+	}
+
+	// Remote
+	// Get Certifications
+	log.Println("Retrieving dependent certifications")
+	err = getter.GetRemoteResources(destination, constants.DefaultCertificationsFolder, opencontrol.GetCertificationsDependencies())
+	if err != nil {
+		return err
+	}
+	// Get Standards
+	log.Println("Retrieving dependent standards")
+	err = getter.GetRemoteResources(destination, constants.DefaultStandardsFolder, opencontrol.GetStandardsDependencies())
+	if err != nil {
+		return err
+	}
+	// Get Components
+	log.Println("Retrieving dependent components")
+	err = getter.GetRemoteResources(destination, constants.DefaultComponentsFolder, opencontrol.GetComponentsDependencies())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+
+// Getter is an interface for how to get and place local and remote resources.
+type Getter interface {
+	GetLocalResources(source string, resources []string, destination string, subfolder string, recursively bool, resourceType constants.ResourceType) error
+	GetRemoteResources(destination string, subfolder string, entries []common.RemoteSource) error
+}
+
+// NewVCSAndLocalGetter constructs a new resource getter with the type of parser to use for the files.
+func NewVCSAndLocalGetter(parser opencontrol.SchemaParser) Getter {
+	return &vcsAndLocalFSGetter{Downloader: NewVCSDownloader(), FSUtil: fs.OSUtil{}, Parser: parser, ResourceMap: mapset.Init()}
```

Comment:
Minor: Mind making this multiple lines? Will be a bit easier to visually scan everything that's being passed in.